### PR TITLE
Add Clima (A100) pipeline

### PR DESCRIPTION
**Pull request by sriharshakandala**
## Purpose 
Add Clima (A100) pipeline

----
- [x] I have read and checked the items on the review checklist.

### CODE REVIEW

File: .buildkite/A100_pipeline/pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+          slurm_mem_per_cpu: 6G
+
+      - label: "GPU unit tests"
+        command: "julia --color=yes --project=test test/runtests.jl"
+        env:
+          CLIMACOMMS_DEVICE: "CUDA"
+        agents:
+          slurm_ntasks: 1
+          slurm_gres: "gpu:1"
+          slurm_exclusive:
```
Comment (Sbozzolo) [inline .buildkite/A100_pipeline/pipeline.yml:51]: This flag does nothing, as you can see from the fact that the builds all happen at the same time, see, e.g, https://buildkite.com/clima/rrtmgp-clima-a100-pipeline/builds/15

And if you want to know on what CUDA device they are run, this is already printed under the "GPUs available on clima" tab.

Comment (sriharshakandala) [reply .buildkite/A100_pipeline/pipeline.yml:51]: I see that, but `nvtop` indicates that each benchmark is running on a different GPU!

Comment (sriharshakandala) [reply .buildkite/A100_pipeline/pipeline.yml:51]: Screenshot 2024-09-17 at 12 17 06 AM

Comment (Sbozzolo) [reply .buildkite/A100_pipeline/pipeline.yml:51]: Yes, `exclusive` is supposed to reserve the entire node for a job. 

The fact that each job runs on a different gpu is not something that needs to be specified.

**Pull request merged**